### ISSUE TITLE
Location Update not sent after app goes to background or when in headless mode

### ISSUE THREAD

**Issue by setiawanh** [transistorsoft/react-native-background-geolocation#735]
## Your Environment
* Plugin version: 3.0.6
* Platform:  Android 
* OS version: 9.0
* Device manufacturer / model: Samsung Note 8
* React Native version (`react-native -v`): react-native-cli: 2.0.1 / react-native: 0.58.4
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
BackgroundGeolocation.ready({
      // common options
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      distanceFilter: 0,

      // android options
      locationUpdateInterval: 15000,
      fastestLocationUpdateInterval: 10000,
      allowIdenticalLocations: true,

      // ARS Options
      stopTimeout: 5,
      disableStopDetection: false, // this is not recommended because will NEVER disable location service until we call background geo location stop

      // http event
      // set once logged in in API.startGPS
      // url: url,
      autoSync: true,
      // set once logged in in API.startGPS
      // headers: {
      //   "token": token,
      //   "Content-Type": "application/json",
      //   "Accept": "application/json"
      // },
      locationTemplate: '{"lat":<%= latitude %>,"lng":<%= longitude %>,"dt":"<%= timestamp %>",uuid:"<%= uuid %>"}',
      // application config
      stopOnTerminate: false,   // <-- Allow the background-service to continue tracking when user closes the app.
      startOnBoot: true,        // <-- Auto start tracking when device is powered-up.
      heartbeatInterval: 60,
      enableHeadless: true, // need to register headless task in the index.js
      reset: true,
      // update notification text
      notification: {
        title: "Initial Notification",
        text: "Initial Text"
      },

      // Application config
      debug: false, // <-- enable this hear sounds for background-geolocation life-cycle.
      logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,



    }, (state) => {
      console.log("- BackgroundGeolocation is configured and ready: ", state.enabled);

      if (!state.enabled) {
        ////
        // 3. Start tracking!
        //
        BackgroundGeolocation.start(function() {
          console.log("- BackgroundGeolocation Start success");
        });

        BackgroundGeolocation.setConfig({
          notification: {
            title: "GPS Ready",
            text: "GPS Ready Text"
          }
        })
      }
    });
  }
```

* When user has signed in, the following are called:
```javascript
    let token = '....';
    let url = this.baseUrl + 'gps';
    BackgroundGeolocation.setConfig({
      url: url,
      headers: {
        "token": token,
        "Content-Type": "application/json",
        "Accept": "application/json"
      },
      // update notification text
      notification: {
        title: "Sending Data",
        text: "Ready to send data."
      }
    }).then((state) => {

    });

    // This handler fires whenever bgGeo receives a location update.
    BackgroundGeolocation.onLocation((location) => {
      console.log('[location] -', location);
      // will call API to send location with the token
      this.sendLocation({ lat: location.coords.latitude, lng: location.coords.longitude, dt: location.timestamp, uuid: location.uuid });
    }, (error) => {
      console.log('[locationError] -', error);
    });


    BackgroundGeolocation.onHttp(response => {
      console.log('http',response);
      console.log('[http] response: ', response.success, response.status, response.responseText);
    });

    BackgroundGeolocation.onHeartbeat((event) => {
      console.log('[onHeartbeat] ', event);

      // You could request a new location if you wish.
      BackgroundGeolocation.getCurrentPosition({
        samples: 1,
        persist: true
      }).then((location) => {
        console.log('[getCurrentPosition] ', location);
        // will call API to send location with the token
        this.sendLocation({ lat: location.coords.latitude, lng: location.coords.longitude, dt: location.timestamp, uuid: location.uuid });
      });
    })
```
* I am using headless for this as well. The following is the headless code in index.js:
```javascript
////
// Define your Headless task -- simply a javascript async function to receive
// events from BackgroundGeolocation:
//
let HeadlessTask = async (event) => {
  let params = event.params;
  console.log('[BackgroundGeolocation HeadlessTask] -', event.name, params);

  switch (event.name) {
    case 'heartbeat':
      // Use await for async tasks
      let location = await getCurrentPosition();
      const api = new API();
      api.sendLocation({ lat: location.coords.latitude, lng: location.coords.longitude, dt: location.timestamp, uuid: location.uuid });
      console.log('sent', { lat: location.coords.latitude, lng: location.coords.longitude, dt: location.timestamp, uuid: location.uuid });
      break;
  }
}

////
// You're free to execute any API method upon BackgroundGeolocation in your HeadlessTask.
// Just be sure to wrap them in a Promise and "await" their completion.
//
let getCurrentPosition = () => {
  return new Promise((resolve) => {
    BackgroundGeolocation.getCurrentPosition({
      samples: 1,
      persist: true
    }).then((location) => {
      resolve(location);
    });
  });
};

////
// Register your HeadlessTask with BackgroundGeolocation plugin.
//
BackgroundGeolocation.registerHeadlessTask(HeadlessTask);
```

## Expected Behavior

1. When the app goes to the background / terminated, it supposes to send GPS data every 60 seconds.
2. When the device is restarted, it should send GPS data.

## Actual Behavior

1. When the app goes to the background / terminated, it keeps sending GPS data but after around 15-20 minutes, it stops sending GPS data. Bring the app back to the foreground, will cause the GPS to be sent again.
2. When the device is restarted, it took around 10 minutes for the GPS data to be sent. And afte 15 - 20 minutes, it stops sending GPS data. Running the app again will cause the GPS to be sent again.

## Context

I am trying to make the app sending the GPS location every minute (or other interval) regardless when the app is in the foreground / background / terminated / and the app is restarted.
</details>

**Comment by setiawanh** [comment]
Hi, anyone experience this? Or any suggestion?

**Comment by christocracy** [comment]
Show me your logs.  See wiki [Debugging](https://github.com/transistorsoft/react-native-background-geolocation/wiki/Debugging)

**Comment by christocracy** [comment]
Btw, if you're only tracking the user "once signed in", you shouldn't be calling `#start` until they're signed in.

```javascript
if (!state.enabled) {
        ////
        // 3. Start tracking!  <-- NO!  Not until they're signed in.  Remove this.
        //
        BackgroundGeolocation.start(function() {
          console.log("- BackgroundGeolocation Start success");
        });
```

**Comment by setiawanh** [comment]
@christocracy  Thank you for the feedback. Sorry for a very delayed response. I have followed your suggestion.

1. I still have `BackgroundGeolocation.ready({ ... })` in the App.js with the exception that it didn't call `#start` when it is completed.

2. I moved `#start` into the user signed in when `#setConfig` is successful.

As for the background location, it stopped when it was headless mode after around 90 minutes. I have attached the log from the `adb logcat -s TSLocationManager`. It stops after 90 minutes and not outputting anything after that.

[log.txt](https://github.com/transistorsoft/react-native-background-geolocation/files/3473430/log.txt)
